packages/server/src/DirectConnection.ts

```diff
@@ -0,0 +1,53 @@
+import { URLSearchParams } from 'url'
+import Document from './Document'
+import type { Hocuspocus } from './Hocuspocus'
+
+export class DirectConnection {
+  documentPromise: Promise<Document> | null = null
+
+  instance!: Hocuspocus
+
+  context: any
+
+  document?: Document
+
+  /**
+   * Constructor.
+   */
+  constructor(
+    documentPromise: Promise<Document>,
+    instance: Hocuspocus,
+    context?: any,
+  ) {
+    this.documentPromise = documentPromise
+    this.instance = instance
+    this.context = context
+  }
+
+  async transact(transaction: (document: Document) => void) {
```

Comment:
not sure I get the idea of this. How would this be used?

```typescript

const server = new Hocuspocus();

const myDocument = await server.getDirectConnection({documentName: 'my-document', context: {}})

await myDocument.transact((doc) => {
  doc.getMap('test').set('a', 'b');
});

// 2nd call would then again trigger addDirectConnection and await the documentPromise?
await myDocument.transact((doc) => {
  doc.getMap('test').set('a', 'b');
});
```

If I didn't get it wrong, I'd probably just make the `await server.getDirectConnection` set up everything (i.e. increase connection count, await document promise). `transact` method would then probably only call `storeDocumentHooks` after the transaction

Reply:
ah, I see you just pushed a fix while I was looking at it, which is doing exactly that :=) resolving!